Sources/Vapor/HTTP/HTTPHeaderCacheControl.swift

```
@@ -0,0 +1,152 @@
+    public var staleWhileRevalidate: Int?
+
+    /// Indicates the client will accept a stale response if the check for a fresh one fails. The value indicates how many *seconds* long the client will accept the stale response after the initial expiration.
+    public var staleIfError: Int?
+
+    /// Initializes the class
+    /// - Parameter rawValue: The initial value.
+    public init(rawValue: Int) {
+        self.rawValue = rawValue
+    }
+
+    private static let exactMatch: [String: Self] = [
+        "must-revalidate": Self.mustRevalidate,
+        "no-cache": Self.noCache,
+        "no-store": Self.noStore,
+        "no-transform": Self.noTransform,
+        "public": Self.isPublic,
+        "private": Self.isPrivate,
+        "proxy-revalidate": Self.proxyRevalidate,
+        "only-if-cached": Self.onlyIfCached,
+        "max-stale": Self.maxStale
+    ]
+
+    private static let prefix: [String: WritableKeyPath<Self, Int?>] = [
+        "max-age": \.maxAge,
+        "s-maxage": \.sMaxAge,
+        "max-stale": \.maxStale,
+        "min-fresh": \.minFresh,
+        "stale-while-revalidate": \.staleWhileRevalidate,
+        "stale-if-error": \.staleIfError
+    ]
+
+    /// Initializes the class
+    /// - Parameter value: The HTTP header string value.
+    public init(stringLiteral value: String) {
+        var set = CharacterSet.whitespacesAndNewlines
+        set.insert(",")
+
+        value
+            .filter { !($0.isWhitespace || $0.isNewline) }
+            .components(separatedBy: set)
+            .forEach {
+                let str = $0.lowercased()
+
+                if let value = Self.exactMatch[str] {
+                    insert(value)
+                    return
+                }
+
+                let parts = str.components(separatedBy: "=")
+                guard parts.count == 2, let keyPath = Self.prefix[parts[0]], let seconds = Int(parts[1]) else {
+                    return
```

Do we want to silently ignore things that don't match? For HTTPCookies and other HTTPHeaders extensions failing to parse everything correctly will typically result in the extension returning nil.

---

General rule is that if you don't recognize an option you just skip it.